src/components/views/elements/ReplyThread.tsx

```diff
@@ -88,7 +88,13 @@ export default class ReplyThread extends React.Component<IProps, IState> {
         // could be used here for replies as well... However, the helper
         // currently assumes the relation has a `rel_type`, which older replies
         // do not, so this block is left as-is for now.
-        const mRelatesTo = ev.getWireContent()['m.relates_to'];
+        //
+        // We're prefer ev.getContent() over ev.getWireContent() to make sure
+        // we grab the latest edit with potentially new relations. But we also
+        // can't just rely on ev.getContent() by itself because historically we
+        // still show the reply from the original message even though the edit
+        // event does not include the relation reply.
+        const mRelatesTo = ev.getContent()['m.relates_to'] || ev.getWireContent()['m.relates_to'];
```

Comment:
Tracing down where this code has gone since this was merged, the codebase has morphed a bit and this code has moved to:

 - [`matrix-js-sdk` -> `src/models/event.ts#L544-L552`](https://github.com/matrix-org/matrix-js-sdk/blob/3870e3395d0caad778ac83eb5ee2daeef53f0e9e/src/models/event.ts#L544-L552) via https://github.com/matrix-org/matrix-js-sdk/pull/2108
    -  `matrix-react-sdk` side: https://github.com/matrix-org/matrix-react-sdk/pull/7565
 - `ReplyThread` was renamed to `ReplyChain` on the `matrix-react-sdk` side
    - [`src/components/views/elements/ReplyChain.tsx`](https://github.com/matrix-org/matrix-react-sdk/blob/b5ece995d2390e8428aa1c2d44dc4ad051d8dc42/src/components/views/elements/ReplyChain.tsx)
    - [`test/components/views/elements/ReplyChain-test.tsx`](https://github.com/matrix-org/matrix-react-sdk/blob/b5ece995d2390e8428aa1c2d44dc4ad051d8dc42/test/components/views/elements/ReplyChain-test.tsx)

Reply:
This is incorrect, we should only ever look at wire content as per the red notice in the spec

![image](https://user-images.githubusercontent.com/2403652/221588579-a9e31423-74b1-4914-ab40-40a0033145d0.png)
![image](https://user-images.githubusercontent.com/2403652/221588630-a3955f87-98dd-4bcb-b2ec-33ac3aa49f8e.png)